MoneyDataType/Currency.cs

```diff
@@ -0,0 +1,136 @@
+
+            var region = new RegionInfo(culture.LCID);
+            if (region is null)
+                throw new ArgumentNullException(nameof(region));
+
+
+            EnglishName = region.CurrencyEnglishName;
+            NativeName = region.CurrencyNativeName;
+            Symbol = region.CurrencySymbol;
+            IsoCode = region.ISOCurrencySymbol;
+            DecimalPlaces = culture.NumberFormat.CurrencyDecimalDigits;
+            _culture = culture;
+        }
+
+        public Currency(string englishName, string nativeName, string symbol, string iSOSymbol, int decimalDigits = 2) 
+        {
+            EnglishName = englishName;
+            NativeName = nativeName;
+            Symbol = symbol;
+            IsoCode = iSOSymbol;
+            DecimalPlaces = decimalDigits;
+            _culture = null;
+        }
+
+        public string Culture => _culture?.Name??IsoCode;
+
+        public int DecimalPlaces { get; }
+
+        public string IsoCode { get; }
+
+        public string Symbol { get; }
+
+        public string EnglishName { get; }
+
+        public string NativeName { get; }
+
+        public bool Equals(ICurrency other) => other != null && IsoCode == other.IsoCode;
+
+        public override bool Equals(object obj) => obj != null && obj is ICurrency other && Equals(other);
+
+        public override int GetHashCode() => IsoCode.GetHashCode();
+
+        public override string ToString() => Symbol;
+
+        public string ToString(decimal amount)
+            => _culture!=null? amount.ToString("C" + DecimalPlaces, _culture) : "(" + (IsoCode ?? "UNK") + ") " + amount.ToString($"N{DecimalPlaces}");
+
+        private string DebuggerDisplay => IsoCode;
+
+        public static bool operator ==(Currency left, Currency right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(Currency left, Currency right)
+        {
+            return !(left == right);
+        }
+
+        public static ICurrency USDollar => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.USD);
+
+        public static ICurrency Euro => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.EUR);
+
+        public static ICurrency Yen => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.JPY);
+
+        public static ICurrency PoundSterling => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.GBP);
+
+        public static ICurrency AustralianDollar => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.AUD);
+
+        public static ICurrency CanadianDollar => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.CAD);
+
+        public static ICurrency SwissFranc => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.CHF);
+
+        public static ICurrency Renminbi => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.CNY);
+
+        public static ICurrency SwedishKrona => KnownCurrencyTable.FromSystemCurrency(SystemCurrency.SEK);
```

Comment:
My intent was to provide the ones you initially defined as "System" ones, similar to how colors have a subset of system colors.  Happy to add all of them but think we should keep the enum as that will make looking them up much easier than a magic string.

Reply:
I had provided those originally because I didn't have time to integrate them all, and didn't think about extracting their data from .NET itself, so I picked the 10 most exchanged. Since we now have all the data, it seems like they should all be exposed. If that's too much work, please don't feel obliged to include it in this PR. We can do it later, just a suggestion if it's not too much inconvenience.

Reply:
All instances added as static instances using ISO names as keys